app/battle_preparations/prepare_to_battle.py

```diff
@@ -0,0 +1,21 @@
+from app.knight.knight_class import Knight
+
+
+def preparations(knights: dict) -> list:
+    knights_instances = [
+        Knight(knights[knight]["name"],
+               knights[knight]["power"],
+               knights[knight]["hp"])
+        for knight in knights
+    ]
+
+    for knight_inst in knights_instances:
+        knight_inst.apply_armor(knights)
+
+    for knight_inst in knights_instances:
+        knight_inst.equip_weapon(knights)
+
+    for knight_inst in knights_instances:
+        knight_inst.use_potion(knights)

```

Comment:
Avoid doing multiple loopings, provide and apply all data during the initialization.